bika/lims/databasesanitize/controller_view.py

```diff
@@ -0,0 +1,63 @@
+import json
+
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from collective.taskqueue.interfaces import ITaskQueue
+from plone.protect import CheckAuthenticator
+from zope.component import queryUtility
+
+from bika.lims import logger
+from bika.lims.browser import BrowserView
+from bika.lims.databasesanitize.analyses import analyses_creation_date_recover
+
+
+class ControllerView(BrowserView):
+    """
+    This class is the controller for the sanitize view. Sanitize view
+    """
+    template = ViewPageTemplateFile("templates/main_template.pt")
+
+    def __call__(self):
+        # Need to generate a PDF with the stickers?
+        if self.request.form.get('analysis_date', '0') == '1':
+            logger.info("Starting sanitation process 'Analyses creation "
```

Comment:
I'd create a function, say `sanitize_analyses_creation_date()` and move there all the code from inside this `if`, so the conditional would look something like (note the 'submit' param -- see the comment https://github.com/naralabs/bika.lims/pull/250/files#r138865101 --):

```python
    def __call__(self):
        if not self.request.form.get('submit', None):
            # The form hasn't been submitted, no tasks required
            return template()
 
        if self.request.form.get('analysis_date', '0') == '1':
            # Fix the creation date of analyses 
            self.sanitize_analyses_creation_date()

    def sanitize_analyses_creation(self):
        logger.info("Starting sanitation process 'Analyses creation'")
        ....
```
This would eventually allow you to allocate new tasks cleanly, like follows:
```python
    def __call__(self):
        if not self.request.form.get('submit', None):
            # The form hasn't been submitted, no tasks required
            return template()
 
        if self.request.form.get('analysis_date', '0') == '1':
            # Fix the creation date of analyses 
            self.sanitize_analyses_creation_date()

        if self.request.form.get('task_2', '0') == '1':
            # Run task_2
            self.task_2()

```